File: CPP_Assignment/Assigment_38/programAQ4.cpp
///////////////////////////////////////////////////////////
//
//  File name : Assignement38Question4.cpp
//  Description :Write generic program to accept N values from user and return largest values.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//
///////////////////////////////////////////////////////////

#include <iostream>
using namespace std;

template <class T>
T Max(T *arr, int iSize)
{
    T iMax = arr[0];

    int i = 0;

    for(i = 0;i < iSize;i++)
    {
        if(arr[i] > iMax)
        {
            iMax = arr[i];
        }
    }
    return iMax;
}

int main()
{
    int arr[] = {10, 20, 30, 40, 50};
    
    float brr[] = {10.0, 3.7, 9.8, 8.7};

    int iRet = Max(arr, 5);
    
    printf("%d\n", iRet); // 50
    
    float fRet = Max(brr, 4);
    
    printf("%f\n", fRet); // 9.8
    
    return 0;
}

File: Java_Assignment/Assignment_53/programAQ4.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement53Question4.java
//  Description : Write a java program which accept two array from user and copy
//                the contents of that array into another array in reverse order and
//                 return new array.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*

*/

import java.util.Scanner;

class MyArray
{
    public int [] CopyArray(int arr[]) 
    {
        int i = 0;

        int Brr[] = new int[arr.length];

        for(i = 0; i < arr.length;i++)
        {
            Brr[i] = arr[i];
        }

        return Brr;
    }
}

public class programAQ4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iSize = 0, i = 0;
        
        System.out.print("Enter size of  array: ");
        iSize = sobj.nextInt();

        int arr[] = new int[iSize];

        System.out.println("Enter elements of  array:");
        for(i = 0; i < iSize; i++)
        {
            arr[i] = sobj.nextInt();
        }

        MyArray aobj = new MyArray();
        
        int iRet [] = aobj.CopyArray(arr);

        System.out.println("Copy Element are :");
        for(i = 0; i < iRet.length;i++)
        {
            System.out.print(iRet[i] + "\t");

        }
        System.out.println();

        sobj.close();
    }   
}